### PR TITLE
Display user's data in the profile setting

### PR DESCRIPTION
**Pull request by shiv-chan**
**What I have done** 👩‍💻 

_I was going to extract the login user's data from all data, but it didn't work well.  I created another slice for the login user's data_
- created `myProfileSlice` to get and store the login user's data
- make the profile setting page display the login user's data

- fix to create a token when signing up
- fix the problem that the previous data would be overwritten and gone when toggling the "available" switch

Please review the code and let me know if you have questions 🙏

### CODE REVIEW

**Comment by miyabitanimchi** [comment]
@shiv-chan sorry maybe you already mentioned about this earlier, but 
_make the profile setting page display the login user's data_
means it automatically displays login user's data in the inputs (in our case it's email?) when a user visits the page?
I logged in and went to the page but I didn't see my email or anything in the inputs so let me make sure what you meant 🙏

スクリーンショット 2021-11-29 午後5 20 44

**Comment by shiv-chan** [comment]
> @shiv-chan sorry maybe you already mentioned about this earlier, but _make the profile setting page display the login user's data_ means it automatically displays login user's data in the inputs (in our case it's email?) when a user visits the page? I logged in and went to the page but I didn't see my email or anything in the inputs so let me make sure what you meant 🙏

@miyabitanimchi This email is going to be public and the user is supposed to set the public email themselves(even though it'll be the same email as the email for logging in).
Probably you haven't specified your **public** email yet. That's why there isn't any input there.

File: client/src/features/profile/setting/ProfileSetting.tsx [hunk 1]
```diff
@@ -5,154 +5,190 @@ import SenpaiProfileSettingItems from './SenpaiProfileSettingItems';
...
-			commaSpliceArr.map((ele) => returnedArr.push(ele.trim()));
-			setInputs({ ...inputs, techStack: returnedArr });
-		} else {
-			setInputs({ ...inputs, [name]: value });
-		}
-	};
+    if (Object.keys(myProfile).length !== 0) {
+      const publicEmail = myProfile.publicEmail
+        ? myProfile.publicEmail
+        : myProfile.email;
```
Comment (miyabitanimchi) [inline client/src/features/profile/setting/ProfileSetting.tsx:34]: We added the conditional here to see if there is `publicEmail` in `myProfile`. If not, we set an email that user used for login.

File: server/controllers/profile.js [hunk 2]
```diff
@@ -1,39 +1,67 @@
...
-				},
-			}
-		).exec();
-		res.status(201).json(updatedUser);
-	} catch (error) {
-		res.status(409).json({ message: error.message });
-	}
+  try {
+    if (!validator.isEmail(publicEmail))
+      throw new Error('Please enter a valid email');
```
Comment (miyabitanimchi) [inline server/controllers/profile.js:11]: We figured put the validator for `publicEmail` which is set in `server/models/userModels` was not working, so we added the code here `server/controllers/profile.js` to validate it.

File: server/models/userModels.js [hunk 3]
```diff
@@ -4,88 +4,93 @@ import validator from 'validator';
...
+        'A password must be at least 6 charactors including at least one letter and number.',
+    },
+  },
+  date: {
+    type: Date,
+    default: new Date(),
+  },
+  name: {
+    type: String,
+    default: '',
```
Comment (miyabitanimchi) [inline server/models/userModels.js:41]: Added a default value for name so we can at least have the key and empty string on the setting page

File: server/models/userModels.js [hunk 4]
```diff
@@ -4,88 +4,93 @@ import validator from 'validator';
...
+      type: String,
+      default: uuidv4(),
+    },
+    isActive: {
+      type: Boolean,
+      default: false,
+    },
+    description: {
+      type: String,
+      default: '',
```
Comment (miyabitanimchi) [inline server/models/userModels.js:63]: Added a default value for description so we can at least have the key and empty string on the setting page

**Review by miyabitanimchi — APPROVED** [review]
The error happened below in my environment in `/profile/setting/kohai` and `/profile/setting/senpai`, and we also found the validator for publicEmail in `server/models/userModels` was not working,  so me and Kaho figured out why and changed some code below with the comments.

スクリーンショット 2021-11-30 午後1 41 49

Thank you for your great work by the way  @shiv-chan 🙌

**Review by Dannyboy222 — APPROVED** [review]
👍

**Pull request merged**